nixos/modules/system/boot/luksroot.nix

```diff
@@ -430,9 +434,22 @@ let
         ${optionalString (lib.versionOlder kernelPackages.kernel.version "5.4") ''
           echo "On systems with Linux Kernel < 5.4, it might take a while to initialize the CRNG, you might want to use linuxPackages_latest."
           echo "Please move your mouse to create needed randomness."
+        ''}
+        ${if dev.fido2.requiresPin then ''
+          mkfifo ${fifo_tmp_fido_pin}
+          read -rsp "FIDO2 pin for ${dev.device}: " fido_pin
+        '' else ''
+          # Dont make a fifo queue or ask for pin since no pin entry is required
         ''}
```

Comment:
```suggestion
        ${lib.optionalString dev.fido2.requiresPin ''
          mkfifo ${fifo_tmp_fido_pin}
          read -rsp "FIDO2 pin for ${dev.device}: " fido_pin
        ''}
```

Reply:
Thank you! I couldn't find a way to do if without else but apparently this is the right way.